### PR TITLE
#1 bootstrap the group module

### PR DESCRIPTION
**Pull request by rezk2ll**
image

### CODE REVIEW

**Review by MichaelBailly — APPROVED** [review]
Khaled: you did a huge work on that one, thank you. It's TL;DR for me, I'm sure other JS developers will give you valuable feedback. Here is mine:

groups got two main features:
* the admin group management pages
* the group resolver, which lookup the group emails when auto-completing on emails, calendar events, ...

What we will need in a short term is the resolver. I'm telling you that so you have it in mind.

File: src/index.test.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+require('./index.js');
+require('angular-mocks/angular-mocks.js');
+require('../test/config/mocks/injector.js');
+require('../test/config/mocks/module.js');
+
+
+var sinonChai = require('sinon-chai/lib/sinon-chai.js');
+var shallowDeepEqual = require('chai-shallow-deep-equal/chai-shallow-deep-equal.js');
+var chaiDatetime = require('chai-datetime/chai-datetime.js');
```
Comment (tuanlc) [inline src/index.test.js:14]: s/var/const/g

Add a blank line after var

File: test/config/mocks/injector.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+angular.module('esn.test.injector', []).run(function($q) {
+  window.$q = $q;
+});
+
+beforeEach(angular.mock.module('esn.test.injector'));
+
+window.$q = angular.injector(["ng"]).get("$q");
```
Comment (tuanlc) [inline test/config/mocks/injector.js:7]:
```suggestion
window.$q = angular.injector(['ng']).get('$q');
```

**Review by tuanlc — COMMENTED** [review]
HUGE WORK :tada: 

It's better if you separate your work by commits. It's easier to follow and review :wink:

File: package.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    "pug-loader": "^2.4.0",
+    "sinon": "1.17.5",
+    "sinon-chai": "2.8.0",
+    "style-loader": "^1.2.1",
+    "svg-inline-loader": "^0.8.2",
+    "webpack": "^4.43.0",
+    "webpack-cli": "^3.3.11"
+  },
+  "scripts": {
+    "serve": "webpack-dev-server --watch --progress",
```
Comment (huy-ta) [inline package.json:33]: Do we need this if this repo is not an SPA?

Comment (rezk2ll) [reply package.json:33]: yep sorry that slipped in, used it to test some stuff

File: src/index.test.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
```
Comment (tuanlc) [inline src/index.test.js:14]:
```suggestion
const sinonChai = require('sinon-chai/lib/sinon-chai.js');
const shallowDeepEqual = require('chai-shallow-deep-equal/chai-shallow-deep-equal.js');
const chaiDatetime = require('chai-datetime/chai-datetime.js');
```

**Review by chamerling — APPROVED** [review]

**Review by huy-ta — APPROVED** [review]

**Pull request merged**